src/test/test-dns-domain.c

```diff
@@ -120,6 +120,37 @@ static void test_dns_name_equal(void) {
         test_dns_name_equal_one("..", "..", -EINVAL);
 }
 
+static void test_dns_name_between_one(const char *a, const char *b, const char *c, int ret) {
+        int r;
+
+        r = dns_name_between(a, b, c);
+        assert_se(r == ret);
+
+        r = dns_name_between(c, b, a);
+        if (ret >= 0)
+                assert_se(r == 0);
+        else
+                assert_se(r == ret);
+}
+
+static void test_dns_name_between(void) {
+        /* see https://tools.ietf.org/html/rfc4034#section-6.1 */
+
+        test_dns_name_between_one("example", "a.example", "yljkjljk.a.example", true);
+        test_dns_name_between_one("a.example", "yljkjljk.a.example", "Z.a.example", true);
+        test_dns_name_between_one("yljkjljk.a.example", "Z.a.example", "zABC.a.EXAMPLE", true);
+        test_dns_name_between_one("Z.a.example", "zABC.a.EXAMPLE", "z.example", true);
+        test_dns_name_between_one("zABC.a.EXAMPLE", "z.example", "\001.z.example", true);
+        test_dns_name_between_one("z.example", "\001.z.example", "*.z.example", true);
+        test_dns_name_between_one("\001.z.example", "*.z.example", "\200.z.example", true);
+        test_dns_name_between_one("*.z.example", "\200.z.example", "example", true);
```

Comment:
Not quite sure if I understand what `dns_name_between()` is supposed to do, but according to RFC4034, Section 6.1, `\200.z.example` does *not* lie in between `*.z.example` and `example` wrt canonical ordering. What am I missing?